xplique/example_based/datasets_operations/harmonize.py

```diff
@@ -0,0 +1,235 @@
+                     targets_dataset: Optional[tf.data.Dataset] = None
+                     ) -> Tuple[tf.data.Dataset, tf.data.Dataset, tf.data.Dataset]:
+    """
+    Splits a TensorFlow dataset into cases, labels, and targets datasets.
+    The dataset is splitted only if it has multiple columns.
+    If the dataset has 2 columns, the second column is assumed to be the labels.
+    If the dataset has several columns but labels and targets are provided,
+    there is a conflict and an error is raised.
+
+    Parameters
+    ----------
+    cases_dataset
+        The dataset to split.
+    labels_dataset
+        Labels associated with the cases in the `cases_dataset`.
+        If this function is called, it should be `None`.
+    targets_dataset
+        Targets associated with the cases in the `cases_dataset`.
+        If this function is called and `cases_dataset` has 3 columns, it should be `None`.
+    
+    Returns
+    -------
+    cases_dataset
+        The dataset used to train the model.
+    labels_dataset
+        Labels associated with the `cases_dataset`.
+    targets_dataset
+        Targets associated with the `cases_dataset`.
+    """
+
+    assert isinstance(cases_dataset, tf.data.Dataset), (
+        f"The dataset should be a `tf.data.Dataset`, got {type(cases_dataset)}."
+    )
+
+    if isinstance(cases_dataset.element_spec, tuple):
+        if len(cases_dataset.element_spec) == 2:
+            assert labels_dataset is None, (
+                "The second column of `cases_dataset` is assumed to be the labels. "\
+                + "Hence, `labels_dataset` should be empty."
+            )
+            labels_dataset = cases_dataset.map(lambda x, y: y)
+            cases_dataset = cases_dataset.map(lambda x, y: x)
+        elif len(cases_dataset.element_spec) == 3:
+            assert labels_dataset is None and targets_dataset is None, (
+                "The second and third columns of `cases_dataset` are assumed to be the labels "\
+                "and targets. Hence, `labels_dataset` and `targets_dataset` should be empty."
+            )
+            targets_dataset = cases_dataset.map(lambda x, y, t: t)
+            labels_dataset = cases_dataset.map(lambda x, y, t: y)
+            cases_dataset = cases_dataset.map(lambda x, y, t: x)
+        else:
+            raise AttributeError(
+                "`cases_dataset` cannot have more than 3 columns, "
+                + f"{len(cases_dataset.element_spec)} were detected."
+            )
+    
+    return cases_dataset, labels_dataset, targets_dataset
+
+
+def harmonize_datasets(
+        cases_dataset: DatasetTensor,
+        labels_dataset: Optional[DatasetTensor] = None,
+        targets_dataset: Optional[DatasetTensor] = None,
+        batch_size: Optional[int] = None,
+    ) -> Tuple[DatasetTensor, DatasetTensor, DatasetTensor, int]:
+    """
+    Harmonizes the provided datasets, ensuring they are either `tf.data.Dataset` or
```

Comment:
Harmonizes the provided datasets, transforming them to `tf.data.Dataset` if necessary. Datasets are also checked in case they are shuffled or do not match in batch_size.